test/core/iomgr/mpmcqueue_test.cc

```diff
@@ -0,0 +1,231 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "src/core/lib/iomgr/executor/mpmcqueue.h"
+
+#include <grpc/grpc.h>
+#include <grpc/support/alloc.h>
+#include <grpc/support/log.h>
+
+#include "src/core/lib/gpr/useful.h"
+#include "src/core/lib/gprpp/thd.h"
+#include "test/core/util/test_config.h"
+
+#define THREAD_SMALL_ITERATION 100
+#define THREAD_LARGE_ITERATION 10000
+
+static void test_no_op(void) {
+  gpr_log(GPR_DEBUG, "test_no_op");
+  grpc_core::MPMCQueue mpmcqueue;
+  gpr_log(GPR_DEBUG, "Checking count...");
+  GPR_ASSERT(mpmcqueue.count() == 0);
+  gpr_log(GPR_DEBUG, "Done.");
+}
+
+// Testing items for queue
+struct WorkItem {
+  int index;
+  bool done;
+
+  WorkItem(int i) : index(i) { done = false; }
+};
+
+static void test_small_queue(void) {
+  gpr_log(GPR_DEBUG, "test_small_queue");
+  grpc_core::MPMCQueue small_queue;
+  for (int i = 0; i < THREAD_SMALL_ITERATION; ++i) {
+    small_queue.Put(static_cast<void*>(new WorkItem(i)));
+  }
+  GPR_ASSERT(small_queue.count() == THREAD_SMALL_ITERATION);
+  // Get items out in FIFO order
+  for (int i = 0; i < THREAD_SMALL_ITERATION; ++i) {
+    WorkItem* item = static_cast<WorkItem*>(small_queue.Get());
+    GPR_ASSERT(i == item->index);
+    delete item;
+  }
+  gpr_log(GPR_DEBUG, "Done.");
+}
+
+static void test_get_thd(void* args) {
```

Comment:
Why not having a "ConsumerThread" to consolidate test_get_thd and test_many_get_thd?